### ISSUE TITLE
PSA: import/export FFDH keys

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#5911]
Implement and test import and export of finite-field Diffie-Hellman (FFDH) keys with a supported group (i.e. the RFC 7919 groups).

This task covers `psa_import_key` (public and private keys), `psa_export_key` (public and private keys) and `psa_export_public_key`, as well as any associated size macro and metadata test associated with `PSA_KEY_TYPE_FFDH`.

Until https://github.com/Mbed-TLS/mbedtls/issues/3261 it will not be possible to exercise FFDH keys. This might mean either adding a temporary exception or doing the two tasks together.

Follow-ups:

* [Key agreement with the FFDH algorithm](https://github.com/Mbed-TLS/mbedtls/issues/3261)
* [Generation of FFDH keys](https://github.com/Mbed-TLS/mbedtls/issues/TODO)
* [Derivation of FFDH keys](https://github.com/Mbed-TLS/mbedtls/issues/3262)

**Comment by mprse** [comment]
Does finite-field Diffie-Hellman (FFDH) corresponds to the following key types in PSA:
- `PSA_KEY_TYPE_DH_PUBLIC_KEY( PSA_DH_FAMILY_RFC7919)` - public FFDH key
- `PSA_KEY_TYPE_DH_KEY_PAIR( PSA_DH_FAMILY_RFC7919)` - public and private FFDH  key ?

(The `FF` is missing in PSA macros)

**Comment by gilles-peskine-arm** [comment]
Yes. I think PSA consistently uses “DH” for the keys and “FFDH” for the algorithm. Which is a bad naming choice now that I notice it, but we're stuck with that now.

**Comment by mprse** [comment]
Found in documentation for export functions:

https://github.com/Mbed-TLS/mbedtls/blob/d86abf2392e88ea116fb3c9a811117185db86b14/include/psa/crypto.h#L703-L709

https://github.com/Mbed-TLS/mbedtls/blob/d86abf2392e88ea116fb3c9a811117185db86b14/include/psa/crypto.h#L781-L785

FFDH is handled in `dhm.h` module:
https://github.com/Mbed-TLS/mbedtls/blob/d86abf2392e88ea116fb3c9a811117185db86b14/include/mbedtls/dhm.h#L114-L127

It looks like in case of `PSA_KEY_TYPE_DH_PUBLIC_KEY( PSA_DH_FAMILY_RFC7919)` we need to export/import `X` and in case of `PSA_KEY_TYPE_DH_KEY_PAIR( PSA_DH_FAMILY_RFC7919)` we need to export/import `GX`.
For export/import it seems that simple copy operation is sufficient?

> For Diffie-Hellman key exchange public keys (key types for which #PSA_KEY_TYPE_IS_DH_PUBLIC_KEY is true), the format is the representation of the public key `y = g^x mod p` as a big-endian byte string. The length of the byte string is the length of the base prime `p` in bytes. 

Regarding the length I assume that the length of `GX` is equal to the length of `P` (and also `X`)?

**Comment by gilles-peskine-arm** [comment]
It's the opposite: X is the private key and GX is the public key. (And GY is the peer's public key: a `mbedtls_dhm_context` is not just a key context, it's an operation context that includes the key.) Both have the same nominal size as P (i.e. they should be represented as the same number of bytes).

**Comment by mprse** [comment]
Yes, sorry I mixed `X` and `GX` (as mentioned in `mbedtls_dhm_context ` above) .

**Comment by mprse** [comment]
>For Diffie-Hellman key exchange key pairs (key types for which #PSA_KEY_TYPE_IS_DH_KEY_PAIR is true), the format is the representation of the private key `x` as a big-endian byte string. The length of the byte string is the private key size in bytes  (leading zeroes are not stripped). 

Why `DH_KEY_PAIR` should keep only the the private key (`X`) and not both private key(`X`) + public key(`GX`)?

**Comment by gilles-peskine-arm** [comment]
The public key can be recalculated from the private key. We use the smallest reasonable format for ECC and DH key pairs to save data size and parsing complexity, so it's just the private value.
